Source/RunActivity/Viewer3D/Weather.cs

```diff
@@ -914,6 +923,94 @@ public AutomaticWeather(Viewer viewer)
         {
         }
 
+        // check weather details, set auto variables
+        void CheckWeatherDetails()
+        {
+            float prevTime = 0;
+
+            foreach (WeatherSetting weatherSet in weatherDetails)
+            {
+                TimeSpan acttime = new TimeSpan((long)(weatherSet.Time * 10000000));
+
+                // check if time is in sequence
+                if (weatherSet.Time < prevTime)
+                {
+                    Trace.TraceInformation("Invalid time value : time out of sequence : {0}", acttime.ToString());
+                    weatherSet.Time = prevTime + 1;
+                }
+                prevTime = weatherSet.Time;
+
+                // check settings
+                if (weatherSet.GetType() == typeof(WeatherSettingOvercast))
```

Comment:
A much better way of doing this is:
```suggestion
                if (weatherSet is WeatherSettingOvercast)
```

Reply:
Replaced all "typeof" checks with "is" statement.